homeassistant/components/template/binary_sensor.py

```diff
@@ -303,11 +303,9 @@ def _update_state(self, result):
             self._delay_cancel()
             self._delay_cancel = None
 
-        state = (
-            None
-            if isinstance(result, TemplateError)
-            else template.result_as_boolean(result)
-        )
+        state: bool | None = None
+        if result is not None and not isinstance(result, TemplateError):
+            state = template.result_as_boolean(result)
```

Comment:
I think we want the corresponding change also in `TriggerBinarySensorEntity`

Reply:
> I think we want the corresponding change also in `TriggerBinarySensorEntity`

Done